src/hud/waiticondisplay.cpp

```diff
@@ -15,87 +18,147 @@ using namespace Impacto::Profile::Dialogue;
 
 void Init() {
-  if (WaitIconCurrentType == +WaitIconType::SpriteAnim) {
-    SpriteAnim = WaitIconSpriteAnim.Instantiate();
-    SpriteAnim.LoopMode = ALM_Loop;
-    SpriteAnim.StartIn();
-  } else if (WaitIconCurrentType == +WaitIconType::SpriteAnimFixed) {
-    SpriteAnim = WaitIconSpriteAnim.Instantiate();
-    SpriteAnim.LoopMode = ALM_Stop;
-    FixedSpriteAnim = static_cast<FixedSpriteAnimation&>(SpriteAnim);
-    FixedSpriteAnim.Def->FixSpriteId = WaitIconFixedSpriteId;
-    FixedSpriteAnim.StartIn();
-  } else {
-    SimpleAnim.DurationIn = WaitIconAnimationDuration;
-    SimpleAnim.LoopMode = ALM_Loop;
-    SimpleAnim.StartIn();
+  if (WaitIconCurrentType == +WaitIconType::None) return;
+
+  switch (WaitIconCurrentType) {
+    case WaitIconType::SpriteAnim:
+      SpriteAnim = WaitIconSpriteAnim.Instantiate();
+      SpriteAnim.LoopMode = ALM_Loop;
+      SpriteAnim.StartIn();
+      break;
+    case WaitIconType::SpriteAnimFixed:
+      SpriteAnim = WaitIconSpriteAnim.Instantiate();
+      SpriteAnim.LoopMode = ALM_Stop;
+      FixedSpriteAnim = static_cast<FixedSpriteAnimation&>(SpriteAnim);
+      FixedSpriteAnim.Def->FixSpriteId = WaitIconFixedSpriteId;
+      break;
+    default:
+      SimpleAnim.DurationIn = WaitIconAnimationDuration;
+      SimpleAnim.LoopMode = ALM_Loop;
+      SimpleAnim.StartIn();
+      break;
   }
 }
 
 void Update(float dt) {
-  if (WaitIconCurrentType == +WaitIconType::SpriteAnim) {
-    SpriteAnim.Update(dt);
-  } else if (WaitIconCurrentType == +WaitIconType::SpriteAnimFixed) {
-    FixedSpriteAnim.Update(dt);
-  } else {
-    SimpleAnim.Update(dt);
+  switch (WaitIconCurrentType) {
+    case WaitIconType::None:
+      return;
+
+    case WaitIconType::SpriteAnim:
+      SpriteAnim.Update(dt);
+      break;
+
+    case WaitIconType::SpriteAnimFixed:
+      FixedSpriteAnim.Update(dt);
+      break;
+
+    default:
+      SimpleAnim.Update(dt);
+      break;
   }
 }
-void Render(glm::vec2 pos, glm::vec4 opacityTint, DialoguePageMode mode) {
-  if (WaitIconCurrentType == +WaitIconType::SpriteAnim) {
-    if (DialogueBoxCurrentType == +DialogueBoxType::CHLCC) {
-      // To deal with multiple DialogueBox
-      glm::vec4 col = glm::vec4(1.0f, 1.0f, 1.0f, opacityTint.a);
-      // Erin DialogueBox
-      if (mode == DPM_REV) {
-        Renderer->DrawSprite(
-            SpriteAnim.CurrentSprite(),
-            glm::vec2(
-                pos.x +
-                    Impacto::Profile::CHLCC::DialogueBox::REVWaitIconOffset.x,
-                pos.y +
-                    Impacto::Profile::CHLCC::DialogueBox::REVWaitIconOffset.y),
-            col);
-
-      } else {
-        Renderer->DrawSprite(
-            SpriteAnim.CurrentSprite(),
-            glm::vec2(pos.x + WaitIconOffset.x, pos.y + WaitIconOffset.y), col);
-      }
+
+static void RenderSpriteAnim(glm::vec2 pos, glm::vec4 opacityTint,
+                             DialoguePageMode mode) {
+  if (GetFlag(Profile::ScriptVars::SF_SHOWWAITICON))
```

Comment:
Moved 'em to `Update`, and also had them only fire `StartIn/Out` when `IsOut/In`.
Don't really see what difference that last one makes, since now it's continuously calling `IsIn/Out`, but I suppose semantically it makes a little more sense

Reply:
Mostly semantics yeah, and I think IsIn/IsOut is basically guaranteed to be inlined with how simple it is, versus StartIn which then calls on a virtual function defined in a cpp (not sure if we have link time optimizations enabled yet)